ideascube/library/views.py

```diff
@@ -66,6 +66,15 @@ def get_context_data(self, **kwargs):
 
 class BookDetail(DetailView):
     model = Book
+    def get(self, request, *args, **kwargs):
+        ret = super().get(request, *args, **kwargs)
+        user = request.user
+        if ( user.is_authenticated()
+         and user.is_staff
```

Comment:
`user.is_staff` implies `user.is_authenticated()`, so you don't need to test both.